widget/playground/src/containers/StyleLayout/StyleLayout.CustomColors.tsx

```diff
@@ -0,0 +1,220 @@
+import type { CustomColorsTypes } from './StyleLayout.types';
+import type { Mode } from '../../store/config';
+import type { WidgetColorsKeys } from '@rango-dev/widget-embedded';
+
+import {
+  ChevronDownIcon,
+  ChevronUpIcon,
+  Collapsible,
+  CustomColorsIcon,
+  Divider,
+  Typography,
+} from '@rango-dev/ui';
+import React, { useEffect, useState } from 'react';
+
+import { ColorItemPicker } from '../../components/ColorItemPicker';
+import { WIDGET_COLORS } from '../../constants';
+import { useConfigStore } from '../../store/config';
+
+import {
+  Color,
+  ColorsContent,
+  CustomColorCollapsible,
+  CustomColors,
+  FieldTitle,
+  Row,
+} from './StyleLayout.styles';
+
+const Colors = (props: { mainColor?: string; secondColor?: string }) => {
+  return (
+    <ColorsContent>
+      <Color style={{ backgroundColor: props.mainColor }} />
+      {props.secondColor !== undefined && (
+        <Color
+          style={{
+            backgroundColor: props.secondColor,
+            position: 'absolute',
+            right: -10,
+          }}
+        />
+      )}
+    </ColorsContent>
+  );
+};
+
+export function CustomColorsSection(props: CustomColorsTypes) {
+  const { tab, selectedPresets } = props;
+  const [openCustomColors, toggleCustomColors] = useState<{
+    tab: Mode;
+    value: boolean;
+  }>({ tab, value: false });
+  const [openCustomColor, setOpenCustomColor] = useState<string | null>(null);
+  const onChangeColors = useConfigStore.use.onChangeColors();
+
+  const { theme } = useConfigStore.use.config();
+  const isDarkTab = tab === 'dark';
+  const isLightTab = tab === 'light';
+  const isAutoTab = tab === 'auto';
+  const isDarkColorsOnly =
+    !!Object.keys(theme?.colors?.dark).length &&
+    !Object.keys(theme?.colors?.light).length;
+  const isLightColorsOnly =
+    !!Object.keys(theme?.colors?.light).length &&
+    !Object.keys(theme?.colors?.dark).length;
```

Comment:
I think you can define `Object.keys(theme?.colors?.light)` & dark or their length in separate variables to avoid repeating